### PR TITLE
fix for a channel int tests

### PR DESCRIPTION
**Pull request by jebrando**

### CODE REVIEW

File: tests/channel_int/channel_int.c [hunk 1]
```diff
@@ -232,19 +234,25 @@ TEST_FUNCTION(test_channel_create_and_destroy)
     /// arrange
 
     /// act
-    THANDLE(CHANNEL) channel = channel_create(g.g_threadpool);
+    THANDLE(THREADPOOL) threadpool = create_threadpool();
```
Comment (dcristoloveanu) [inline tests/channel_int/channel_int.c:237]: >  THANDLE(THREADPOOL) thre [](http://example.com/codeflow?start=2&length=26)

This is part of arrange, so it should be placed appropriately in the test.

File: tests/channel_int/channel_int.c [hunk 2]
```diff
@@ -232,19 +234,25 @@ TEST_FUNCTION(test_channel_create_and_destroy)
...
+
+    THANDLE(CHANNEL) channel = channel_create(threadpool);
 
     /// assert
     ASSERT_IS_NOT_NULL(channel);
 
     // cleanup
     THANDLE_ASSIGN(CHANNEL)(&channel, NULL);
+    threadpool_close(threadpool);
+    THANDLE_ASSIGN(THREADPOOL)(&threadpool, NULL);
```
Comment (dcristoloveanu) [inline tests/channel_int/channel_int.c:247]: Isn't the fact that we're creating a threadpool for each test hiding a deeper issue?
I am not getting how this is solving the crash actually ...

**Review by dcristoloveanu — CHANGES_REQUESTED** [review]
:clock1:

File: tests/channel_int/channel_int.c (hunk 2 above)
Comment (jebrando) [reply tests/channel_int/channel_int.c]: Fixed, this is also spawned a new task here [https://msazure.visualstudio.com/One/_workitems/edit/25500968](https://msazure.visualstudio.com/One/_workitems/edit/25500968)

File: build/devops_gated.yml [hunk 3]
```diff
@@ -122,7 +122,7 @@ jobs:
 - job: linuxubuntu
   displayName: 'Build Linux Ubuntu'
   pool:
-    name: Azure-MessagingStore-LinuxBuildMachinePool
+    name: Azure-MsgStore-Linux2204BuildMachinePool
```
Comment (dcristoloveanu) [inline build/devops_gated.yml:125]: >me: Azure-MsgStore-Linux2204BuildMachinePool [](http://example.com/codeflow?start=6&length=44)

Hmm, is this intended?

Comment (dcristoloveanu) [reply build/devops_gated.yml:125]: This seems to be from another change? Or is it needed for this test to pass?

**Review by dcristoloveanu — APPROVED** [review]
:shipit:

File: build/devops_gated.yml (hunk 3 above)
Comment (jebrando) [reply build/devops_gated.yml:125]: The test fails more on the new version of linux, new helgrind I'm guessing.

**Pull request merged**